File: SinHilos1.java
package Java;
// Este programa simula la ejecución secuencial de tareas para comparar el tiempo con hilos

public class SinHilos1 {

    // Método que simula una tarea secuencial
    public static void imprimirMensaje(String nombre, int veces) {
        for (int i = 0; i < veces; i++) {
            System.out.println(nombre + ": Ejecutando iteración " + (i + 1));
            try {
                Thread.sleep(1000); // Pausa la ejecución por 1 segundo
            } catch (InterruptedException e) {
                System.out.println(nombre + " fue interrumpido.");
            }
        }
    }

    public static void main(String[] args) {
        // Inicia el contador de tiempo
        long inicio = System.currentTimeMillis();

        // Ejecutar las funciones de forma secuencial
        imprimirMensaje("Tarea 1", 3); // Ejecuta "Tarea 1"
        imprimirMensaje("Tarea 2", 2); // Luego ejecuta "Tarea 2"

        // Detiene el contador de tiempo
        long fin = System.currentTimeMillis();

        // Calcula el tiempo total
        double tiempoTotal = (fin - inicio) / 1000.0;

        System.out.printf("Todas las tareas han terminado. Tiempo total: %.2f segundos.%n", tiempoTotal);
    }
}

File: SinHilos1.py
#Mismo programa de Hilos1 solo que no se usan hilos para comparar tiempo de ejecucion
import time  # Importa el módulo time para manejar pausas en la ejecución y medición del tiempo

# Definimos una función que será ejecutada secuencialmente
def imprimir_mensaje(nombre, veces):  # La función recibe un nombre y cuántas veces ejecutar el bucle
    for i in range(veces):            # Repite el bucle `veces` veces
        print(f"{nombre}: Ejecutando iteración {i + 1}")  # Imprime el nombre y la iteración actual
        time.sleep(1)  # Pausa la ejecución por 1 segundo para simular una tarea lenta

# Inicia el contador de tiempo
inicio = time.time()

# Ejecutar las funciones de forma secuencial
imprimir_mensaje("Tarea 1", 3)  # Ejecuta la función para "Tarea 1"
imprimir_mensaje("Tarea 2", 2)  # Luego ejecuta la función para "Tarea 2"

# Detiene el contador de tiempo
fin = time.time()

# Calcula el tiempo total
tiempo_total = fin - inicio

print(f"Todas las tareas han terminado. Tiempo total: {tiempo_total:.2f} segundos.")  # Imprime el tiempo total
